### ISSUE TITLE
AGS4: Broken room editor at lower color depth

### ISSUE THREAD

**Issue by AlanDrake** [adventuregamestudio/ags#2349]
**Describe the bug**
Background editor doesn't work properly with color depth != 32, trying to assign a sprite to an object will result in a `System.ArgumentOutOfRangeException: `

This is probably related to that user having trouble with 8bit mode a few months ago.

```
Error: System.ArgumentOutOfRangeException: The requested range exceeds the end of the array.
   in System.Runtime.InteropServices.Marshal.CopyToNative(Object source, Int32 startIndex, IntPtr destination, Int32 length)
   in System.Runtime.InteropServices.Marshal.Copy(Byte[] source, Int32 startIndex, IntPtr destination, Int32 length)
   in AGS.Editor.BitmapExtensions.SetRawData(Bitmap bmp, Byte[] rawData, PixelFormat pixelFormat) in E:\Projects\ags\ags-draconian\Editor\AGS.Editor\Utils\BitmapExtensions.cs:riga 268
   in AGS.Editor.BitmapExtensions.SetRawData(Bitmap bmp, Byte[] rawData) in E:\Projects\ags\ags-draconian\Editor\AGS.Editor\Utils\BitmapExtensions.cs:riga 277
   in AGS.Editor.ObjectsEditorFilter.Paint(Graphics graphics, RoomEditorState state) in E:\Projects\ags\ags-draconian\Editor\AGS.Editor\Panes\Room\RoomEditFilters\ObjectsEditorFilter.cs:riga 168
   in AGS.Editor.RoomSettingsEditor.bufferedPanel1_Paint(Object sender, PaintEventArgs e) in E:\Projects\ags\ags-draconian\Editor\AGS.Editor\Panes\Room\RoomSettingsEditor.cs:riga 403
   in System.Windows.Forms.Control.OnPaint(PaintEventArgs e)
   in System.Windows.Forms.Control.PaintWithErrorHandling(PaintEventArgs e, Int16 layer)
   in System.Windows.Forms.Control.WmPaint(Message& m)
   in System.Windows.Forms.Control.WndProc(Message& m)
   in System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   in System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   in System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   in System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

**AGS Version**
AGS 4.0

**To Reproduce**
1. Create empty game
2. Set color depth to 8 or 16 bit
3. Import a sprite
4. Create a room
5. Create an object and assign the imported sprite

**Expected behavior**
It should work normally.

**Comment by ivan-mogilko** [comment]
There's also #2218, but I am not sure if this is a duplicate?

**Comment by ivan-mogilko** [comment]
Confirmed fixed by #2354
